core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -73,8 +76,13 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
   var numCompletedStages = 0
   var numFailedStages = 0
 
+  // Executors:
+  val executors = ListBuffer[ExecutorUIData]()
```

Comment:
It doesn't make sense for the `JobProgressListener` to keep track of executor data. Can you put these data structures in the in `ExectuorsListneer` instead? It's fine if you need to add a reference to that listener so it's available here.